apps/studio/src/server/modules/resource/resource.router.ts

```diff
@@ -572,4 +574,23 @@ export const resourceRouter = router({
         (a, b) => resourceIds.indexOf(a.id) - resourceIds.indexOf(b.id),
       )
     }),
+
+  getIndexPage: protectedProcedure
+    .input(getIndexPageSchema)
+    .output(getIndexPageOutputSchema)
```

Comment:
clarification - this specifies the return type of this procedure and shouldn't be needed normally because trpc can infer the return type.

could i check why we need to type hint for `parent` here? i'd prefer that we just let `trpc` handle the typing tbh but this isn't a big deal

Reply:
@seaerchin not much reasons actually, was just thinking of using the schema as a self-documentation!